Adding some more route53 resources

zerbitx opened the pull request:
Internally we needed these resources nuked.  Hopefully they're of use to you as well.

zerbitx committed:
feat: adding route53 resolver resource

Co-authored-by: Tom Neyland <tcneyland@gmail.com>
Co-authored-by: Max Krieg <maxwell.krieg@gmail.com>

zerbitx committed:
Merge pull request #3 from pluralsight/route53-resolvers

feat: adding route53 resolver resource

Commit:
feat: adding route53 traffic policies

Co-authored-by: Ryan Swanson <ryan.a.swanson@gmail.com>
Co-authored-by: Russell Centanni <russell.centanni@gmail.com>

zerbitx committed:
Merge branch 'master' of github.com:pluralsight/aws-nuke

zerbitx committed:
fix: fixing traffic policy and instance deletion

der-eismann commented on resources/route53-resolver-endpoints.go:
```diff
@@ -0,0 +1,90 @@
+}
+
+func init() {
+	register("Route53ResolverEndpoints", ListRoute53ResolverEndpoints)
```
```suggestion
	register("Route53ResolverEndpoint", ListRoute53ResolverEndpoints)
```

We always use the singular for a resource.

der-eismann commented on resources/route53-resolver-endpoints.go:
```diff
@@ -0,0 +1,90 @@
+	output, err := svc.ListResolverEndpoints(params)
+
+	if err != nil {
+		return resources, err
```
```suggestion
		return nil, err
```

der-eismann commented on resources/route53-resolver-endpoints.go:
```diff
@@ -0,0 +1,90 @@
+			})
+
+		if err != nil {
+			return resources, err
```
```suggestion
			return nil, err
```

der-eismann commented on resources/route53-resolver-endpoints.go:
```diff
@@ -0,0 +1,90 @@
+		resources = append(resources, resolverEndpoint)
+	}
+
+	return resources, err
```
```suggestion
	return resources, nil
```

der-eismann commented on resources/route53-resolver-endpoints.go:
```diff
@@ -0,0 +1,90 @@
+			return resources, err
+		}
+
+		for _, ip := range ipsOutput.IpAddresses {
+			resolverEndpoint.ips = append(resolverEndpoint.ips, &route53resolver.IpAddressUpdate{
+				Ip:       ip.Ip,
+				IpId:     ip.IpId,
+				SubnetId: ip.SubnetId,
+			})
+		}
```
What do you actually need these for? They are not used in the output or remove function.

zerbitx replied:
Ah, left in from initial dev when we did have them in the output for a time.  I'll take them out, thanks!

der-eismann commented on resources/route53-resolver-rules.go:
```diff
@@ -0,0 +1,114 @@
+)
+
+func init() {
+	register("Route53ResolverRules", ListRout53ResolverRules)
```
```suggestion
	register("Route53ResolverRule", ListRoute53ResolverRules)
```

der-eismann commented on resources/route53-resolver-rules.go:
```diff
@@ -0,0 +1,114 @@
+}
+
+// ListRout53ResolverRules produces the resources to be nuked.
+func ListRout53ResolverRules(sess *session.Session) ([]Resource, error) {
```
```suggestion
func ListRoute53ResolverRules(sess *session.Session) ([]Resource, error) {
```

zerbitx replied:
Good eye!

der-eismann commented on resources/route53-resolver-rules.go:
```diff
@@ -0,0 +1,114 @@
+	output, err := svc.ListResolverRules(&route53resolver.ListResolverRulesInput{})
+
+	if err != nil {
+		return resources, err
```
```suggestion
		return nil, err
```

der-eismann commented on resources/route53-resolver-rules.go:
```diff
@@ -0,0 +1,114 @@
+	associationsOutput, err := svc.ListResolverRuleAssociations(&route53resolver.ListResolverRuleAssociationsInput{})
+
+	if err != nil {
+		return resources, err
```
```suggestion
		return nil, err
```

der-eismann commented on resources/route53-resolver-rules.go:
```diff
@@ -0,0 +1,114 @@
+		vpcIds     []*string
+	}
+
+	association struct {
+		id    *string
+		vpcID *string
+	}
```
AFAICS this struct isn't used anywhere, is it?

zerbitx replied:
Nope, good catch.

der-eismann commented on resources/route53-resolver-rules.go:
```diff
@@ -0,0 +1,114 @@
+// Properties provides debugging output
+func (r *Route53ResolverRule) Properties() types.Properties {
+	return types.NewProperties().
+		Set("Id", r.id).
```
```suggestion
		Set("ID", r.id).
```

der-eismann commented on resources/route53-traffic-policies.go:
```diff
@@ -0,0 +1,113 @@
+	return instances, nil
+}
+
+type Route53TrafficPolicy struct {
+	svc       *route53.Route53
+	id        *string
+	name      *string
+	version   *int64
+	instances []*route53.TrafficPolicyInstance
+}
```
Please put this at the top like in the other resources.

der-eismann commented on resources/route53-traffic-policies.go:
```diff
@@ -0,0 +1,113 @@
+func (tp *Route53TrafficPolicy) Properties() types.Properties {
+	return types.NewProperties().
+		Set("ID", *tp.id).
+		Set("NAME", *tp.name)
```
```suggestion
		Set("Name", *tp.name)
```

der-eismann commented:
Closes #508

zerbitx committed:
Code review changes and NextToken checks (#508)

* Removed unused types
* Renamed resources to singular form
* Updated return values
* Updated loops to keep going until NextToken is nil

zerbitx committed:
Code review change spelling fix (#508)

der-eismann commented on resources/route53-traffic-policies.go:
```diff
@@ -0,0 +1,113 @@
+	return types.NewProperties().
+		Set("ID", *tp.id).
+		Set("Name", *tp.name)
+}
```
Sorry, completely missed this the first time. Could you please add a `String()` function for traffic policies?

der-eismann reviewed (dismissed):
Everything else is looking good now. Thanks for the changes!

zerbitx committed:
Moving passed
 • roles that can be listed, but not retrieved.
 • role policies that that fail to list.
